### DISCUSSION TITLE
Duplicate Floppy Disk with Datapack Recipe?

### DISCUSSION THREAD

**Discussion by Flexico** [cc-tweaked/CC-Tweaked#1878] [General]
While converting my survival world from Fabric to Forge, I unintentionally duplicated a couple of disks, so that there are multiple items that point to the same folder. I soon discovered this is extremely useful, and I'm wondering if there's a way to make a recipe using a datapack to accomplish this legitimately, similarly to the way you can duplicate smithing templates.

To be clear, I'm not asking for this to be added to CC:T itself, just something for me to play around with.

**Comment by Lupus590** [comment]
I know that you can do it with computers using `/giveitem` or whatever the command is, you have to specify the id as nbt data or something. I would assume that you can use the same trick with disks.

**Comment by SquidDev** [comment]
Unfortunately this isn't as easy as one might hope, as vanilla's recipe system doesn't allow you do dynamic outputs (i.e. NBT/components), at least without writing Java code.

That said, on Minecraft 1.20.6+, CC does come with some dynamic recipe support. I've not tested this recipe, but something like this might work:

```json
{
  "type": "computercraft:transform_shapeless",
  "category": "redstone",
  "ingredients": [
    {"item": "minecraft:paper"},
    {"item": "minecraft:redstone"},
    {"item": "computercraft:disk"}
  ],
  "result": {
    "count": 2,
    "id": "computercraft:disk"
  },
  "function": [
    {
      "type": "computercraft:copy_components",
      "from": {"item": "computercraft:disk"}
    }
  ]
}
```
